### ISSUE TITLE
Prometheus Query Editor: Obstructs keyboard navigation

### ISSUE THREAD

**Issue by tskarhed** [grafana/grafana#35378]
**Steps to reproduce**:
1. Go to Explore
2. Choose a Prometheus datasource
3. Navigate to the query field with tab

**Actual Result**:
A tab is put in the field

**Expected Result**
Some option to continue to navigate with tab. Discussions in Slack has come with a couple of suggestions, `esc` would exit edit mode. Toggling tab-trapper behaviour with a global `Ctrl+M`

**Relevant WCAG Criteria:** 2.1.2 WCAG Criterion

**Comment by tskarhed** [comment]
Modifying Slate's behaviour might be the solution here, which could also solve accessibility issues in other places.
